src/components/calcite-time-picker/calcite-time-picker.tsx

```diff
@@ -0,0 +1,702 @@
+        case "ArrowUp":
+          event.preventDefault();
+          this.incrementMinute();
+          break;
+      }
+    }
+  };
+
+  private secondKeyDownHandler = (event: KeyboardEvent): void => {
+    if (numberKeys.includes(event.key)) {
+      const keyAsNumber = parseInt(event.key);
+      if (this.second.startsWith("0")) {
+        const secondAsNumber = parseInt(this.second);
+        if (secondAsNumber > 5) {
+          this.setTime("second", keyAsNumber);
+        } else {
+          this.setTime("second", `${secondAsNumber}${keyAsNumber}`);
+        }
+      } else {
+        this.setTime("second", keyAsNumber);
+      }
+    } else {
+      switch (event.key) {
+        case "Backspace":
+          this.setTime("second", "--");
+          break;
+        case "ArrowDown":
+          event.preventDefault();
+          this.decrementSecond();
+          break;
+        case "ArrowUp":
+          event.preventDefault();
+          this.incrementSecond();
+          break;
+      }
+    }
+  };
+
+  private setTime = (
+    key: "hour" | "minute" | "second" | "ampm",
+    value: number | string | AmPm
+  ): void => {
+    this.timeChanged = true;
+    switch (key) {
+      case "hour":
+        this.hour = typeof value === "number" ? formatNumberAsTimeString(value) : value;
+        break;
+      case "minute":
+        this.minute = typeof value === "number" ? formatNumberAsTimeString(value) : value;
+        break;
+      case "second":
+        this.second = typeof value === "number" ? formatNumberAsTimeString(value) : value;
+        break;
+      case "ampm":
+        if (this.hour === "--") {
+          this.ampm = value as AmPm;
+        } else {
+          const hourAsNumber = parseInt(this.hour);
+          switch (value) {
+            case "AM":
+              if (hourAsNumber >= 12) {
+                this.hour = formatNumberAsTimeString(hourAsNumber - 12);
+              }
+              break;
+            case "PM":
+              if (hourAsNumber < 12) {
+                this.hour = formatNumberAsTimeString(hourAsNumber + 12);
+              }
+              break;
+          }
+          this.ampm = value as AmPm;
+        }
+        break;
+    }
+  };
+
+  // --------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  // --------------------------------------------------------------------------
+
+  connectedCallback() {
+    if (this.hourDisplayFormat === "12") {
+      this.ampm = this.getAmPm();
+    }
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  render(): VNode {
+    const iconScale = this.scale === "s" || this.scale === "m" ? "s" : "m";
+    const includeSeconds = this.step !== 60 ? true : false;
+    return (
+      <Host>
+        <div class="time-picker">
+          <div>
+            <span
```

Comment:
Can you use a button element here?

Reply:
@eriklharper was there a reason a button element couldn't be used here? It would be more semantic if it could be used.